lib/ForkProcessSandbox.cpp

```diff
@@ -0,0 +1,48 @@
+
+#include "ForkProcessSandbox.h"
+
+#include "TestResult.h"
+
+#include <sys/mman.h>
+#include <unistd.h>
+
+Mutang::ExecutionResult Mutang::ForkProcessSandbox::run(std::function<void (ExecutionResult *)> Func) {
+
+  void *SharedMemory = mmap(NULL,
+                            sizeof(ExecutionResult),
+                            PROT_READ | PROT_WRITE,
+                            MAP_SHARED | MAP_ANONYMOUS,
+                            -1,
+                            0);
+
+  ExecutionResult *SharedResult = new (SharedMemory) ExecutionResult();
+
+  int ChildPid = fork();
+  if (ChildPid == -1) {
+    exit(1);
+  }
+
+  else if (ChildPid == 0) {
+    Func(SharedResult);
+    exit(0);
+  }
+
+  else {
+    /// TODO: Handle at least 3 cases:
+    /// 1) Child process crashes
+    /// 2) Child process exits 1
+    /// 3) Child process hangs
```

Comment:
> We can make many assumptions regarding potential hanging mutants an workaround them in advance, but imo it's better to face them in reality and then solve.
> 
> How did you overcome this problem?

Forcing the test frameworks that would execute the tests against such infinite runtime mutations (thats the name I came up with) to have a timeout. But not the mutation testing engine itself.

The idea is that the the following done by hand, should be quivalent to what the mutation testing engine does:

* Insert the code by hand
* Run the subset of the tests the mutation testing engine would run

can result in the same "kill reason" for being observable. If the test framework has a timeout, the user can reproduce the "kill reason" by hand, but if the mutation testing engine on "top of the test framework" has a timeout, the user cannot realistically reproduce the same kill reason, on more complex cases this can lead to significant amounts of time wasted.

Hence my call for: Tests should enforce timeouts, not mutation testing engines.